### PR TITLE
Implement undo/redo in view editor

### PR DESCRIPTION
**Pull request by elvisisking**
See commits for more details

### CODE REVIEW

**Comment by elvisisking** [comment]
Reopening to try and figure out what the merge conflicts are.

File: src/main/ngapp/src/app/dataservices/shared/view.model.ts [hunk 1]
```diff
@@ -105,6 +104,75 @@ export class View {
...
+   * @param {SchemaNode} sourceToAdd the source being added
+   */
+  public addSource( sourceToAdd: SchemaNode ): void {
+    const index = this.sources.findIndex( ( source ) => source.getName() === sourceToAdd.getName() );
+
+    if ( index === -1 ) {
+      this.sources.push( sourceToAdd );
+    }
+  }
+
```
Comment (mdrillin) [inline src/main/ngapp/src/app/dataservices/shared/view.model.ts:119]: getName() is not adequate for identifying the source, but I think you are aware of that - and we can change to path with subsequent commit

**Review by mdrillin — APPROVED** [review]
I'm fine to merge if you want.  We can update once the path changes are merged.

**Review by phantomjinx — APPROVED** [review]
Looks fine.

**Pull request merged**